### PR TITLE
Extended help section for standard webpack

### PR DESCRIPTION
**Pull request by bigopon**
*Doc*: (setup-webpack.md): Update doc for setup with standard webpack configuration
  * File changed:
  
  [Setup-webpack.md](https://github.com/aurelia/framework/blob/master/doc/article/en-US/setup-webpack.md)

This gives some basic guide line for standard webpack configuration, enabling flexibility in modifying the configs.

This is a result of as the orginial author refused to make a pull request
* aurelia/skeleton-navigation#688
* **NOTE**:
    - Breaking changes in webpack `2.1.0.beta-23`, that no longer allow custom properties on config

### CODE REVIEW

**Comment by nashwaan** [comment]
@bigopon That's an excellent documentation. Well done.

You may also want to add in section **8) Reminder** another point to:
`npm install --save-dev less-loader`.

File: doc/article/en-US/setup-webpack.md [hunk 1]
```diff
@@ -85,3 +85,390 @@ Integration tests are performed with [Protractor](http://angular.github.io/protr
...
+    const hasProcessFlag = flag => process.argv.join('').indexOf(flag) > -1;
+    const path = require('path');
+    const webpack = require('webpack');
+    const HtmlWebpackPlugin = require('html-webpack-plugin');
+    const CopyWebpackPlugin = require('copy-webpack-plugin');
+    const AureliaWebpackPlugin = require('aurelia-webpack-plugin');
+    const WebpackMd5Hash = require('webpack-md5-hash');
+    const DefinePlugin = require('webpack/lib/DefinePlugin');
+    const ExtractTextPlugin = require('extract-text-webpack-plugin');
+    const cssnano = require('cssnano');
```
Comment (nashwaan) [inline doc/article/en-US/setup-webpack.md]: `cssnano` is not used anywhere in this file.

File: doc/article/en-US/setup-webpack.md [hunk 2]
```diff
@@ -85,3 +85,390 @@ Integration tests are performed with [Protractor](http://angular.github.io/protr
...
+
+  ```js
+    const hasProcessFlag = flag => process.argv.join('').indexOf(flag) > -1;
+    const path = require('path');
+    const webpack = require('webpack');
+    const HtmlWebpackPlugin = require('html-webpack-plugin');
+    const CopyWebpackPlugin = require('copy-webpack-plugin');
+    const AureliaWebpackPlugin = require('aurelia-webpack-plugin');
+    const WebpackMd5Hash = require('webpack-md5-hash');
+    const DefinePlugin = require('webpack/lib/DefinePlugin');
```
Comment (bigopon) [inline doc/article/en-US/setup-webpack.md]: Fixed. Thanks for pointing that out. I also commented out cssnano, but still leave it there just as reminder

Comment (boazblake) [inline doc/article/en-US/setup-webpack.md]: @bigopon thanks for writing this! I really appreciate the markup.... following your instructions I noticed an error with webpack build: 2.1.0.beta-23 should be 2.1.0-beta.23 #626

Comment (bigopon) [inline doc/article/en-US/setup-webpack.md]: It's official webpack version, idk why they named it like that though

Comment (bigopon) [inline doc/article/en-US/setup-webpack.md]: Yeah just realized that, fixed 👍 

Also changed webpack version to beta.25

Comment (boazblake) [inline doc/article/en-US/setup-webpack.md]: thats what it was! ... 
however, still faced with html-minifier-loader issues: 

` Module not found: Error: Can't resolve 'html-minifier-loader'`
but even 
` npm install html-minifier html-minifier-loader --save`

still resulted in:
`ERROR in ./~/html-minifier/~/uglify-js/tools/node.js`

Comment (bigopon) [inline doc/article/en-US/setup-webpack.md]: Yeah sorry, I fixed that after your comment, shoulda pointed out. I'll check the `html-minifier-loader`.  Don't know why it works for me

Comment (bigopon) [inline doc/article/en-US/setup-webpack.md]: I guess that's some loaders loading content asynchronously:
  * pugjs/pug-loader#8

A suggested fix for that is to add:
  ```js
    node: {
      fs: "empty"
    },
  ```
to your base config object.

@boazblake I also updated the first section about dependencies to make it clearer

**Comment by nashwaan** [comment]
May i suggest to use
```js
const metadata = {
    ...
    HMR: process.argv.join('').indexOf('hot') >= 0 || !!process.env.WEBPACK_HMR
```
and delete the very first line:
```js
const hasProcessFlag = flag => process.argv.join('').indexOf(flag) > -1;
```

**Comment by bigopon** [comment]
@nashwaan  Fixed 👍 . I was not sure if any other flags should be used so I extracted it to a separate function

@boazblake I also added a section to guide how to use aurelia plugins. Hope it helps

**Comment by boazblake** [comment]
@nashwaan @bigopon 

Thank you for your help!

So far I am still getting the same `'fs'` error... even with copy and paste... so it must  be something else I have done  - If it works for the two of you I would image its all good though.
Ill take a look at the guides for the plugins and make sure I am doing it correctly - Iv only been coding since March of this year so I am more than likely making more than a couple of errors.

**Comment by nashwaan** [comment]
@bigopon It looks to me all the `JSON.stringify()` used in `new webpack.DefinePlugin({` are redundant. 

In Section 9, point 3:
`3. Start your project to check if the plugin is propery config`. => properly configured.

Apart from this typo, point 3 provides a workaround if the **aurelia** plugin itself has improper `package.json` configuration. While the information you provided is very valuable, I find it weird to put that in the official aurelia documentation and ask the user to put the fix in their `package.json`. I think it will be more convenient if you inform the maintainers of the aurelia plugins to properly configure their `package.json`, unless these plugins are not the official aurelia plugins. In this case, I wonder if this information is available for the plugin authors.

**Comment by bigopon** [comment]
Typo fixed.

I don't think it's the best idea to put in official document about plugins, but future plugin may have that problem and i think it's quite necessary to let other developers know. What do you suggest @nashwaan ?

**Comment by nashwaan** [comment]
Ideally, the user should not be bothered about properly configuring `aurelia.build.resources` in his  `package.json` in case the `package.json` of the plugin is not doing that right. This is very inconvenient and many users are likely to miss/forget about this workaround. It would be great if you can do pull requests in the `package.json` for the plugins that you are aware of that are having this issue. The idea is to avoid shifting the burden to the user side as much as possible.

Probably, the source of the problem is we don't have **A Plugin Author** section in the docs as [mentioned here](https://github.com/aurelia/documentation/issues/181). I guess that's why such a problem exists in the first place and quite possibly will grow more. Until a doc for developing a plugin is provided, then I think it is ok to keep the information you mentioned in section 9.3 as a workaround solution.

**Comment by nashwaan** [comment]
@bigopon 
Looking at `webpack.config.js` file, I find it intimidating for new users. Yes, you did a great job to make the file achieve the best for a professional aurelia project. But I would suggest to provide a simpler version (minimal) that helps to get aurelia properly running for a new project. After this simple version, you can put the professional version either as a whole or as a diff. What do you think?

**Comment by nashwaan** [comment]
@bigopon In webpack 2, It is no longer necessary to manually _eat up_ the string after the extension in `test`. For example, `(\?\S*)?` portion in the following line:
```
                test: /\.(png|jpe?g|gif|svg|eot|woff|woff2|ttf)(\?\S*)?$/,
```
Can be safely removed:
```
                test: /\.(png|jpe?g|gif|svg|eot|woff|woff2|ttf)$/,
```

**Comment by nashwaan** [comment]
@bigopon , you mentioned in the **NOTE** of this page:

> During development, loading page will have splash message appear without style for a short period but it should be fine for production build as all style will be extracted to a css file and loaded before any content.

Another solution will be (from [Webpack from First Principles](https://www.youtube.com/watch?v=WQue1AN93YU) see at 13:10) to ensure the css, which transformed into javascript, is loaded in the `<head>` tag rather than the `<body>` which is the behavior of `HtmlWebpackPlugin` by default.
I.e:
```js
        new HtmlWebpackPlugin({
            inject: 'head', // to ensure css styles are inserted before showing the body
            title: title,
            template: 'index.html',
            chunksSortMode: 'dependency'
        }),
```

And if this solves the problem, will this make `extract-text-webpack-plugin` unnecessary?

**Comment by boazblake** [comment]
@nashwaan @bigopon 
I went back over my steps and it appears that even though I had `npm install` after pasting in the dependencies, nothing had actually happened and all I had needed to do was explicitly install `html-webpack-plugin`. 
Good job with the guide!

**Comment by bigopon** [comment]
@boazblake glad that it worked for you

@nashwaan I separated the guide into basic and advanced sections. It does look simpler to start with and more beginner friendly. All of your suggestions have been added. For loading style in a `style.min.css`, I think it's better to have css loaded in a separate file than together with javascript.

**Comment by boazblake** [comment]
@bigopon @nashwaan 

Would  you happen to know why the app no longer 'sees' the location of my svgs and jpegs?
Many thanks to you guys!

**Comment by nashwaan** [comment]
Looking at the basic version. Hmmm, I was thinking of even simpler version. A minimal version with no gadgets, no debug, no optimizations, and not much comment.

More preciesly:
```js
const path = require('path');
const webpack = require('webpack');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const AureliaWebpackPlugin = require('aurelia-webpack-plugin'); 
const project = require('./package.json');

module.exports = {
    entry: {
        app: [], // <-- this array will be filled by the aurelia-webpack-plugin
        aurelia: Object.keys(project.dependencies).filter(dep => dep.startsWith('aurelia-'))
    },
    output: {
        path: path.resolve('dist'),
        filename: '[name].bundle.js',
    },
    module: {
        rules: [
            {
                test: /\.js$/,
                exclude: /node_modules/, // or include: path.resolve('src'),
                loader: 'babel-loader',
                query: {
                    presets: ['es2015', 'stage-1'],
                    plugins: ['transform-decorators-legacy']
                }
            },
            {
                test: /\.html$/,
                exclude: /index\.html$/, // index.html will be taken care by HtmlWebpackPlugin
                loader: 'html-loader'
            },
            {
                test: /\.css$/, 
                loaders: ['style-loader', 'css-loader']
            },
            {
                test: /\.(png|jpe?g|gif|svg|eot|woff|woff2|ttf)$/,
                loader: 'url-loader'
            }
        ]
    },
    plugins: [
        new webpack.ProvidePlugin({
            regeneratorRuntime: 'regenerator-runtime', // to support await/async syntax
            Promise: 'bluebird', // because Edge browser has slow native Promise object
            jQuery: 'jquery', // because 'bootstrap' by Twitter depends on jQuery
        }),
        new AureliaWebpackPlugin({
            root: path.resolve(),
            src: path.resolve('src'),
            baseUrl: '/'
        }),
        new HtmlWebpackPlugin({
            template: 'index.html',
        }),
        new webpack.optimize.CommonsChunkPlugin({ // to eliminate code duplication across bundles
            name: ['aurelia']
        })
    ]
};
```
Even preferably without `HtmlWebpackPlugin` but could not figure out how to do this.

The aim is to make it more digestible for newcomers so they would understand what is minimally required to get aurelia running. More of a `hello world` webpack config for aurelia.

I would also prefer to remove `ejs` syntax from `index.html`
```html
<!DOCTYPE html>
<html>
  <head>
    <title>Aurelia Webpack Skeleton</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <base href="/">
    <!-- imported CSS are concatenated and added automatically -->
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
  </head>
  <body aurelia-app="main">
    <div class="splash">
      <div class="message">Aurelia Webpack Skeleton</div>
      <i class="fa fa-spinner fa-spin"></i>
    </div>
    <!-- Uncomment below line for Webpack Dev Server auto reload -->
    <!--<script src="/webpack-dev-server.js"></script>-->
  </body>
</html>
```

Again, the idea is to throw less technologies in the beginning.

**Comment by lares83** [comment]
Hi,
I tried your guildlines and after that I got this error
```
ERROR in Template execution failed: ReferenceError: development is not defined

ERROR in   ReferenceError: development is not defined
```
After some investigation I found bug in your code namely in 
```
new webpack.DefinePlugin({
              '__DEV__': true,
              'ENV': metadata.ENV,
              'HMR': metadata.HMR,
              'process.env': {
                  'ENV': metadata.ENV,
                  'NODE_ENV': metadata.ENV,
                  'HMR': metadata.HMR,
                  'WEBPACK_HOST': metadata.host,
                  'WEBPACK_PORT': metadata.port
              }
          })
```
there should be stringified ENV and WEBPACK values so it should be sth like this:

```
new webpack.DefinePlugin({
            '__DEV__': true,
            'ENV': JSON.stringify(metadata.ENV),
            'HMR': metadata.HMR,
            'process.env': {
                'ENV': JSON.stringify(metadata.ENV),
                'NODE_ENV': JSON.stringify(metadata.ENV),
                'HMR': metadata.HMR,
                'WEBPACK_HOST': JSON.stringify(metadata.host),
                'WEBPACK_PORT': JSON.stringify(metadata.port)
            }
        })
```

After that everything works fine

**Comment by bigopon** [comment]
@nashwaan I'm not sure if merging Aurelia & bootstrap chunks would be ok. All of official guide seems to separate them.

@lares83 Haven't tried to run after removing `JSON.stringify` 😄

**Comment by nashwaan** [comment]
@bigopon  I think you can safely remove this line from `new AureliaWebpackPlugin()` as it is not mentioned anywhere in [aurelia-webpack-plugin](https://github.com/aurelia/webpack-plugin)
```js
              baseUrl: '/'
```
@bigopon From my understanding  aurlia-bootstrap chunk is used for faster loading of the initial app rather than waiting for all aurelia modules to be processed. I merged them here for simpler `webpack.config.js` as optimizations should not be the first thing to learn. Maybe @niieani can shed more lights on this.

**Comment by nashwaan** [comment]
@bigopon @niieani My last comment is to add the full version of the comprehensive `webpack.config.js` ([before simplification](https://github.com/aurelia/framework/pull/626/files/8faa46ef75b0282808f45d7353a8b85151a75d95?short_path=996ae31#diff-996ae31e3231929c56d9541219cc7e93)) in a final section **Suggested Production Setup**. Because some of the good debug and optimization codes are dropped from the current document. Also, it will help to see how all the extra code are put together.

Apart from this, I think the doc is ready for merge.

**Comment by boazblake** [comment]
@lares83 : your suggestion solved my issue! thank you!

@bigopon : 
I think you left something out here

`Modify our config in webpack.config.js by replacing:`

```
{
    test: /\.html$/,
    exclude: /index\.html$/, // index.html will be taken care by HtmlWebpackPlugin
    use: [
        'raw-loader',
        'html-minifier-loader'
    ]
}
```
With:

 ```
 {
      test: /\.html$/,
      exclude: /index\.html$/, // index.html will be taken care by HtmlWebpackPlugin
      use: [
          'raw-loader',
          'html-minifier-loader'
      ]
  }
```

File: doc/article/en-US/setup-webpack.md [hunk 3]
```diff
@@ -85,3 +85,388 @@ Integration tests are performed with [Protractor](http://angular.github.io/protr
...
+
+5. Adjust development tasks<br/>
+  Modify the dev task so it doesn't throw an error,
+  by replacing old dev task in `package.json` 
+  with the slightly different version (without `--progress`)<br/>
+    Replace:
+
+    ```json
+    "server:dev": "cross-env NODE_ENV=development node ./node_modules/webpack-dev-server/bin/webpack-dev-server --inline --progress --profile --watch",
+    "server:dev2": "cross-env NODE_ENV=development node ./node_modules/webpack-dev-server/bin/webpack-dev-server --inline --progress --profile --watch",
```
Comment (niieani) [inline doc/article/en-US/setup-webpack.md]: What is `server:dev2`? We don't currently have such NPM script.

File: doc/article/en-US/setup-webpack.md [hunk 4]
```diff
@@ -85,3 +85,388 @@ Integration tests are performed with [Protractor](http://angular.github.io/protr
...
+                  'stage-1'
+              ],
+              plugins: ['transform-decorators-legacy']
+          }
+      }
+      ```
+
+4. **Using plugins**
+  * Plugins registry repository: [Plugins](https://github.com/aurelia/registry)
+  * To bundle plugins' dependencies properly, all sub modules of a plugin has to be put in `aurelia.build.resources` in either that plugin's `package.json` or your project's `packag.json`
```
Comment (niieani) [inline doc/article/en-US/setup-webpack.md]: typo:

> or your project's `packag.json` 

=> 

> or your project's `package.json`

File: doc/article/en-US/setup-webpack.md [hunk 5]
```diff
@@ -85,3 +85,388 @@ Integration tests are performed with [Protractor](http://angular.github.io/protr
...
+  This is crucial but not all aurelia plugins were aware of this matters / or built before this standard configuration. If you want to use a plugin, follow these steps:
+
+  1. Install a plugin like normal. Ex. `npm install aurelia-dialog --save`
+  2. Go to your project `package.json`, look for path `"aurelia.build.resources"`, add plugin's module name (ex. "aurelia-dialog") to `resources` array
+  3. Start your project to check if the plugin is properly configured
+      * If webpack doesn't complain anything, plugin is good
```
Comment (niieani) [inline doc/article/en-US/setup-webpack.md]: `complain anything` => `complain`

**Comment by bigopon** [comment]
@niieani @nashwaan 

Some typos fixed, I also added *suggested production setup* section

File: doc/article/en-US/setup-webpack.md [hunk 6]
```diff
@@ -85,3 +85,555 @@ Integration tests are performed with [Protractor](http://angular.github.io/protr
...
+            },
+            'css-loader',
+            'less-loader' // <--- This was added to enable "import 'style.less'"
+        ]
+    },
+    ```
+
+3. **Javascript optimization**.
+
+  * To diliver a smaller bundle for production. Add to your `plugins` in the configuration:
```
Comment (niieani) [inline doc/article/en-US/setup-webpack.md]: diliver -> deliver

**Comment by lares83** [comment]
@boazblake no problem :)
@bigopon @niieani  there is one more bug namely when I launch 
``` npm run build:prod ```
i received this error
```
Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.entry['app'] should be one of these:
   string | [string]
   The entry point for one output file.
 - configuration.entry should be one of these:
   object { <key>: string | [string] } | string | [string]
   The entry point(s) of the compilation.
```
it looks like in webpack.beta25 there can't be an empty array in an entry 
``` entry: {
        app: [], // <-- this array will be filled by the aurelia-webpack-plugin
        aurelia: Object.keys(project.dependencies).filter(dep => dep.startsWith('aurelia-'))
    },
```

**Comment by niieani** [comment]
@lares83 I've opened an issue https://github.com/webpack/webpack/issues/3123

**Comment by lares83** [comment]
@niieani great thanks :)

**Comment by nashwaan** [comment]
@niieani In the initial `webpack.config.js`, there are two entry junks for aurelia:
```js
const aureliaBootstrap = [
    'aurelia-bootstrapper-webpack',
    'aurelia-polyfills',
    'aurelia-pal-browser',
    'regenerator-runtime'
];
const aureliaModules = Object.keys(project.dependencies).filter(dep => dep.startsWith('aurelia-'));
module.exports = {
    entry: {
        'app': [],
        'aurelia-bootstrap': aureliaBootstrap,
        'aurelia': aureliaModules.filter(pkg => aureliaBootstrap.indexOf(pkg) === -1)
    },
```

In the simplified version, we used this only:
```js
module.exports = {
    entry: {
        'app': [],
        'aurelia': Object.keys(project.dependencies).filter(dep => dep.startsWith('aurelia-'))
    },
```

But, for a production setup, does the first version provide faster initial loading of aurelia? Aren't  `aurelia-bootstrap` chunk and `aurelia` chunk injected in the `index.html` at the same time?

**Comment by niieani** [comment]
@bigopon @nashwaan using `project.dependencies` is incomplete though, because those dependencies have their dependencies too, so some Aurelia packages may end up in the app chunk if imported from there too.

Separating aurelia-bootstrap and aurelia entrypoints was necessary before the most recent changes to the way bootstrapper-webpack works, because of the way PAL was initialized. Right now it _should_ work without the additional chunk, but I've not tested it. They're injected into `index.html` at the same time, but they're loaded synchronously, which means bootstrapper packages gets loaded first.

**Comment by EisenbergEffect** [comment]
Wow. Tons of collaboration going on here. Great work all! Where are we on this? Ready to go or needs  more iterations?

**Comment by nashwaan** [comment]
@EisenbergEffect Generally, I guess the doc is ready. Thanks for the excellent efforts by @bigopon.

The only one point that I am still not sure about is the comment by @niieani 

> using `project.dependencies` is incomplete though...

I assume @niieani  suggests to revert back to manual loading of aurelia modules:
```js
const coreBundles = {
  bootstrap: [
    'aurelia-bootstrapper-webpack',
    'aurelia-polyfills',
    'aurelia-pal',
    'aurelia-pal-browser',
    'regenerator-runtime',
    'bluebird'
  ],
  aurelia: [
    'aurelia-bootstrapper-webpack',
    'aurelia-binding',
    'aurelia-dependency-injection',
    'aurelia-event-aggregator',
    'aurelia-framework',
    'aurelia-history',
    'aurelia-history-browser',
    'aurelia-loader',
    'aurelia-loader-webpack',
    'aurelia-logging',
    'aurelia-logging-console',
    'aurelia-metadata',
    'aurelia-pal',
    'aurelia-pal-browser',
    'aurelia-path',
    'aurelia-polyfills',
    'aurelia-route-recognizer',
    'aurelia-router',
    'aurelia-task-queue',
    'aurelia-templating',
    'aurelia-templating-binding',
    'aurelia-templating-router',
    'aurelia-templating-resources'
  ]
}
```

rather than using `project.dependencies` as shown in my previous comment. If manual loading is more accurate, then lets use it in the final `webpack.config.js` file.

**Comment by nashwaan** [comment]
@EisenbergEffect Also, it seems that there is an issue as reported by @lares83  regarding empty array `[]` in the entry chunks. This is an issue in the latest beta version of webpack 2 (`webpack.beta25`) and @niieani has reported it webpack/webpack#3123. Now it is pending with the maintainer of webpack project.

**Comment by niieani** [comment]
@nashwaan that's right. We should revert back to manually listing all Aurelia modules to be included in the `aurelia` bundle. We probably don't need the bootstrap entrypoint though. A workaround for the webpack issue could be using `['./src/main']` instead of `[]` which should be added to `app` in any case.

**Comment by nashwaan** [comment]
@bigopon As advised by @niieani , I suggest to do following changes in the final `webpack.config.js`:

```js
const aureliaModules: [
    'aurelia-bootstrapper-webpack',
    'aurelia-binding',
    'aurelia-dependency-injection',
    'aurelia-event-aggregator',
    'aurelia-framework',
    'aurelia-history',
    'aurelia-history-browser',
    'aurelia-loader',
    'aurelia-loader-webpack',
    'aurelia-logging',
    'aurelia-logging-console',
    'aurelia-metadata',
    'aurelia-pal',
    'aurelia-pal-browser',
    'aurelia-path',
    'aurelia-polyfills',
    'aurelia-route-recognizer',
    'aurelia-router',
    'aurelia-task-queue',
    'aurelia-templating',
    'aurelia-templating-binding',
    'aurelia-templating-router',
    'aurelia-templating-resources'
];

module.exports = {
    entry: {
        'app': ['./src/main'],
        'aurelia': aureliaModules
    },
```

Note: not tested.

**Comment by bigopon** [comment]
@nashwaan @niieani @EisenbergEffect 
This all credit goes to @nashwaan with his original: aurelia/skeleton-navigation#668

Fixed for last suggestion. I could go further with Javascript mangling section in last section, by adding a list of private properties in aurelia core modules to help reduce bundle size a lot more, but wasn't sure if it should be.

**Comment by lares83** [comment]
@bigopon I found a small mistake in docs there should be 
```const aureliaModules = [``` instead of  ```const aureliaModules: [```

**Comment by bigopon** [comment]
@lares83 I was thinking the same, but i think for start

**Comment by bigopon** [comment]
Accdidentally hit close & comment

@lares83 I think for very basic version, having aurelia core bundled into aurelia is ok. I think @nashwaan has a good point here is that keep this section extremely simple so anyone can start without feeling intimidating. We have a last section for production build, with things done right so probably it won't be a problem.

**Comment by lares83** [comment]
@bigopon I see your point, thanks for explanation :)

I wondering about one more thing, do we still need bluebird?
As far as I know slow promises in Edge have already been fixed in Edge 14
(https://wpdev.uservoice.com/forums/257854-microsoft-edge-developer/suggestions/11329845-improve-promise-implementation-performance?tracking_code=a6dbd08f77bf7c9b26693bac96309a81)

@EisenbergEffect @niieani what do you think about it?

**Comment by lares83** [comment]
@niieani I'm testing this webpack configuration and I stubmle across one issue namely when I setup in
```package.json```
```
"aurelia": {
    "build": {
      "resources": [
        {
            "path": "users",
            "bundle": "users",
            "lazy": true
        }
      ]
    }
  },
``` 
and launch ```npm run prod:build``` I should receive sth like this 
```
0.fe2de0b647382fef77d8.chunk.js    2.07 kB       0  [emitted]  users
app.9c05dcbfe6ae9cb85b5f.bundle.js     32.9 kB       1  [emitted]  app
aurelia.6a3f85872c3812bc64f5.bundle.js     460kB       2  [emitted]  aurelia
```

but I get only 
```
app.e80a0814f88eb6feb289.bundle.js  32.9 kB       0  [emitted]  app
aurelia.4c55aae59e21d975bf8b.bundle.js   460 kB    1  [emitted]  aurelia
```
and additionally these warnings

```
WARNING in ./src ^\.\/.*$
Module not found: Error: Can't resolve 'bundle' in 'G:\aurelia\skeleton\src'
 @ ./src ^\.\/.*$
 @ ./~/aurelia-loader-webpack/dist/native-modules/aurelia-loader-webpack.js
 @ multi aurelia

WARNING in ./src ^\.\/.*$
Module not found: Error: Can't resolve 'bundle' in 'G:\aurelia\skeleton\src'
 @ ./src ^\.\/.*$
 @ ./~/aurelia-loader-webpack/dist/native-modules/aurelia-loader-webpack.js
 @ multi aurelia
```

**Comment by niieani** [comment]
@lares83 is there an issue for the problem you're describing?

**Comment by lares83** [comment]
@niieani I haven't seen an issue reported for this problem. 
It works fine with a current version of skeleton-esnext-webpack (webpack 2.1.0-beta.22), so it could be another issue with webpack 2.1.0-beta.25

**Comment by peterver** [comment]
I have no clue where to report this, sorry if this is not the correct place :3 

After switching from the \@easy-webpack config to the regular one i started getting issues with the presets for babel (namely : stage-1 and es2015 not being found),

In case anyone else was having this issue, replacing **babel-env** with **babel-preset-stage-1** and **babel-preset-es2015** does the trick.

**Pull request merged**